ouroboros-consensus/src/Ouroboros/Consensus/Mempool/TxSeq.hs

```diff
@@ -58,6 +58,7 @@ newtype TxSeq tx = TxSeq (FingerTree TxSeqMeasure (TxTicket tx))
 instance Foldable TxSeq where
   foldMap f (TxSeq txs) = Foldable.foldMap (f . (\(TxTicket tx _) -> tx)) txs
   null      (TxSeq txs) = Foldable.null txs
+  length    (TxSeq txs) = mSize $ FingerTree.measure txs
```

Comment:
Yeah, needed for O(1) size which you use elsewhere. Good catch.

Alternatively (or also), export a special `size` from this module. People will expect `TxSeq.size` to be O(1) whereas `length` is expected to be often O(n), and forces the reader to think about (or check) if it's O(1) in this case.